sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -2006,13 +2008,18 @@ def convert_pipeline_outputs_to_dict(
     output name to PipelineChannel."""
     if pipeline_outputs is None:
         return {}
+    elif isinstance(pipeline_outputs, dict):
+        # This condition is required to support pipelines that return NamedTuples.
```

Comment:
Is this not covered by `isinstance(..., tuple)`?

Reply:
Good call out. This comment was not clear enough. I just pushed up a [commit](https://github.com/kubeflow/pipelines/pull/11196/commits/17e0c5d9a358a251934fa24fd28e9f2298547570) that expands on it. Some of the nested pipeline tests with namedtuples that we added to the `subdagio` directory were failing until we added this handling to the IR compiler.

Reply:
[Here's](https://github.com/kubeflow/pipelines/blob/17e0c5d9a358a251934fa24fd28e9f2298547570/samples/v2/subdagio/multiple_parameters_namedtuple.py) a specific example. If you try to compile IR from this file, it raises the following error:

```
  return component_factory.create_component_from_func(
Traceback (most recent call last):
  File "/Users/alex/code/pipelines/validation.py", line 21, in <module>
    def core() -> NamedTuple('outputs', val1=str, val2=str):  # type: ignore
  File "/Users/alex/code/pipelines/sdk/python/kfp/dsl/pipeline_context.py", line 71, in pipeline
    return component_factory.create_graph_component_from_func(
  File "/Users/alex/code/pipelines/sdk/python/kfp/dsl/component_factory.py", line 708, in create_graph_component_from_func
    return graph_component.GraphComponent(
  File "/Users/alex/code/pipelines/sdk/python/kfp/dsl/graph_component.py", line 71, in __init__
    pipeline_spec, platform_spec = builder.create_pipeline_spec(
  File "/Users/alex/code/pipelines/sdk/python/kfp/compiler/pipeline_spec_builder.py", line 1915, in create_pipeline_spec
    pipeline_outputs_dict = convert_pipeline_outputs_to_dict(pipeline_outputs)
  File "/Users/alex/code/pipelines/sdk/python/kfp/compiler/pipeline_spec_builder.py", line 2022, in convert_pipeline_outputs_to_dict
    raise ValueError(f'Got unknown pipeline output: {pipeline_outputs}')
ValueError: Got unknown pipeline output: {'val1': {{channel:task=core-comp;name=val1;type=String;}}, 'val2': {{channel:task=core-comp;name=val2;type=String;}}}
```

Adding the `dict` conditional resolved this error. Shout out to @zazulam for identifying / correcting it.